realm/realm-library/src/main/cpp/io_realm_internal_OsSet.cpp

```diff
@@ -188,6 +205,208 @@ Java_io_realm_internal_OsSet_nativeRemoveString(JNIEnv* env, jclass, jlong set_p
+        jsize values_size = values.size();
+        bool set_has_changed = false;
+        for (int i = 0; i < values_size; i++) {
+            JStringAccessor value = values[i];
+            JavaAccessorContext context(env);
+            const std::pair<size_t, bool>& insert_pair = set.insert(context, Any(value));
+
+            // If we get true it means the element was not there and therefore it has changed
+            if (insert_pair.second) {
+                set_has_changed = true;
+            }
+        }
+        return set_has_changed;
+    }
+    CATCH_STD()
+    return false;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_io_realm_internal_OsSet_nativeAsymmetricDifference(JNIEnv*,
+                                                        jclass,
+                                                        jlong set_ptr,
+                                                        jlong other_set_ptr) {
+    auto& set = *reinterpret_cast<realm::object_store::Set*>(set_ptr);
+    auto& other_set = *reinterpret_cast<realm::object_store::Set*>(other_set_ptr);
+
+    // If other set is a subset of set it means set will change after the difference
+    bool has_changed = other_set.is_subset_of(set);
+    set.assign_difference(other_set);
+    return has_changed;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_io_realm_internal_OsSet_nativeRemoveAllString(JNIEnv* env, jclass, jlong set_ptr,
+                                                   jobjectArray j_values) {
+    try {
+        auto& set = *reinterpret_cast<realm::object_store::Set*>(set_ptr);
+        JObjectArrayAccessor<JStringAccessor, jstring> values(env, j_values);
+        jsize values_size = values.size();
+        bool set_has_changed = false;
+        for (int i = 0; i < values_size; i++) {
+            JStringAccessor value = values[i];
+            JavaAccessorContext context(env);
+            const std::pair<size_t, bool>& remove_pair = set.remove_any(Mixed(StringData(value)));
+
+            // If we get true it means the element was not there and therefore it has changed
+            if (remove_pair.second) {
+                set_has_changed = true;
+            }
+        }
+        return set_has_changed;
+    }
+    CATCH_STD()
+    return false;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_io_realm_internal_OsSet_nativeIntersect(JNIEnv*,
+                                             jclass,
+                                             jlong set_ptr,
+                                             jlong other_set_ptr) {
+    auto& set = *reinterpret_cast<realm::object_store::Set*>(set_ptr);
+    auto& other_set = *reinterpret_cast<realm::object_store::Set*>(other_set_ptr);
+
+    // If other set intersects set it means set will not change after the intersection
+    bool has_changed = !set.intersects(other_set);
+    set.assign_intersection(other_set);
+    return has_changed;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_io_realm_internal_OsSet_nativeRetainAllString(JNIEnv* env, jclass, jlong set_ptr,
```

Comment:
Maybe worth escaping to mixed prior to this so there is a chance of not having to do these for all the types!?? 😓 
Or maybe even get to a point where we could just use some core built-in generic operation instead